File: time_insert.c
#include <stdio.h>
#include "hash.h"
#include <stdlib.h>
#include <assert.h>
#include <float.h>

int main(int argc, char *argv[])
{
    // Timing test for N elements
    
    // Default size = 17
    
    if (argc == 5) {
              
        double size = atoi(argv[1]);
        int check_flag = atoi(argv[2]);
        int delete_flag = atoi(argv[3]);
        int starting_size = atoi(argv[4]);


        Hash * H = createHash(1, starting_size);

        if (!size || size > DBL_MAX) {
            fprintf(stderr, "Use size between 0->DOUBLE_MAX\n");
            return EXIT_FAILURE;
        }        
        
        for (int i = 0; i < size; i++) {
            put(H, i, 1);
        }
        
        if (check_flag) {
            for (int i = 0; i < size; i++) {
                assert(1 == get(H, i));
            }
        }

        printf("Total items = %d\n", H->num_elem);

        if (delete_flag) {
            for (int i = 0; i < size; i++) {
                del(H, i);
            }
        }

        free_hash(H);

    } else {
        fprintf(stderr, "Usage ./time_insert <size> <check_flag> <delete_flag> <starting_size>\n");
    }

    return EXIT_SUCCESS;
}

File: time_insert.py
#!/usr/bin/python

# Test speed against time_insert.c

import sys
import random as rand
import time

H = {}

if (len(sys.argv) == 4):

    size = int(sys.argv[1])
    check_flag = int(sys.argv[2])
    delete_flag = int(sys.argv[3])
    
    if (size < 1):
        print "Use size > 1"
        sys.exit(1)
        
    H = {}
        
    for i in range(0, size):
        H[i] = 1
    
    if (check_flag):
        for i in range(0, size):
            assert H[i] == 1
    
    if (delete_flag):
        for i in range(0, size):
            H.pop(i, None)

else:
    print "Usage ./resize_test <size> <check_flag> <delete_flag> "


